tests/README.md

```diff
@@ -0,0 +1,173 @@
+
+## :page_with_curl: Terms and Concepts
+
+#### Main Workflow/Pipeline
+
+Every pipeline that exists in the [pipelines/](../pipelines/) directory of WARP can be though of as a "main" workflow. These are the components that are actually being tested. 
+
+#### Wrapper Workflow
+
+Every main pipeline has an accompanying "wrapper" workflow in the [verification/test-wdls/](../verification/test-wdls/) directory. These wrapper workflows are designed to run the main workflow and pass its outputs along to the validation workflow.
+
+#### Validation Workflow
+
+The validation workflow takes the output of a completed pipeline and compares it against a known truth set for that pipeline. As with the wrapper workflow, each main workflow must also have a [validation workflow](../verification/).
+
+#### Plumbing vs Scientific data
+
+Each main workflow should have two sets of test data, plumbing and scientific. The plumbing test data is typically a set of downsampled inputs designed to test that the pipeline runs to completion without error. The scientific test data is a set of full-sized inputs designed to thoroughly test the pipeline.
+
+These test inputs exist alongside each pipeline in their respective `test_inputs` directory.
+
+#### Truth Branches
+
+The testing framework has a concept of "truth branches". This allows users to define a branch of truth output to run against.
+
+In most cases the branch being run against is "master" which corresponds to a google storage bucket containing the agreed upon output of the pipeline as it exists in the "master" git branch.
+
+#### Results vs Storage buckets
+
+Every workflow run in cromwell has its output stored in an "execution" bucket. When running the pipeline tests it's important that we store this ouptut in a location that we can later reference if needed.
+
+All test workflows that are run will have their output copied to a "results" bucket -> `gs://broad-gotc-test-results/`
+
+When we want to update the "truth" branches for our pipelines we can copy that output to the respective "truth" bucket -> `gs://broad-gotc-test-storage/`
+
+## :hammer_and_pick: Testing Process
+
+We use Jenkins to run our test suite with two webhooks to kick off tests: PR to dev branch kicks off plumbing tests, PR to main/master kicks off scientific tests.
+
+Jenkins will run a job called `Smart Tests` which can be viewed in the [dsp-jenkins](https://github.com/broadinstitute/dsp-jenkins/blob/master/jobs/gotc-jenkins/WarpSmartTestJob.groovy) repo.
+
+The job of the smart tests is to run the `get_changed_pipeline_worklow_test_args.sh` bash script. The script performs a git diff against the target branch and recursively searches for any pipelines that have been changed either directly or indirectly via an imported sub-workflow.
+
+For each pipeline that has changed it will prepare the arguments for the Scala tool that submits the wrapper workflow to cromwell.
```

Comment:
```suggestion
For each pipeline that has changed it will prepare the arguments for the Scala tool that submits the wrapper workflow to Cromwell.
```